pytato/utils.py

```diff
@@ -87,3 +89,56 @@ def with_indices_for_broadcasted_shape(val: prim.Variable, shape: ShapeType,
         return val
     else:
         return val[get_indexing_expression(shape, result_shape)]
+
+
+def extract_dtypes(exprs: List[Union[Array, Number]]) -> List[DtypeOrScalar]:
+    dtypes: List[DtypeOrScalar] = []
+    for expr in exprs:
+        if isinstance(expr, Array):
+            dtypes.append(expr.dtype)
+        else:
+            assert isinstance(expr, Number)
+            dtypes.append(expr)
+
+    return dtypes
+
+
+def broadcasted_binary_op(a1: Union[Array, Number], a2: Union[Array, Number],
```

Comment:
```suggestion
def broadcast_binary_op(a1: Union[Array, Number], a2: Union[Array, Number],
```
(Merriam Webster says "broadcasted" is also OK, but I'm more familiar with "broadcast".)

Reply:
Sure. 6566d74a9d818b3ea9b423d9c4c8c6fbd2c83c14